packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1136,6 +1151,18 @@ export interface BucketProps {
    * @default - No inventory configuration
    */
   readonly inventories?: Inventory[];
+  /**
+   * The objectOwnership of the bucket.
+   *
+   *  if array contains BucketOwnerPreferred it will add ownershipControls rule with objectOwnership = BucketOwnerPreferred
+   *
+   *  if array contains ObjectWriter it will add ownershipControls rule with objectOwnership = ObjectWriter
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/about-object-ownership.html
+   *
+   * @default - No ObjectOwnership configuration, uploading account will own the object.
+   *
+   */
+  readonly objectOwnerships?: ObjectOwnership[];
```

Comment:
Yes it is for future-proofing see the linked issue #11591

Reply:
I see, we don't normally opt for future proofing in the CDK. Also i'm not sure how this structure will allow for future proofing without breaking changes since the array element type will have to change.

I think we should stick to a single value at the moment and figure it out later when we actually see how the API looks like.

Reply:
Sure. I will update the code